slick/src/main/scala/slick/ast/Node.scala

```diff
@@ -385,6 +385,16 @@ final case class ForUpdate(generator: TermSymbol, from: Node) extends ComplexFil
   protected[this] def rebuildWithSymbols(gen: ConstArray[TermSymbol]) = copy(generator = gen(0))
 }
 
+/** A .headOption call. */
+final case class HeadOption(from: Node) extends SimplyTypedNode with UnaryNode {
```

Comment:
But why not use `FirstOption` for both? There's no rule that says a `ClientSideOp` can't be on the server side, the trait only means that we prefer to move it to the client side when it occurs at the outer layer.